Basic Hono docs

gewenyu99 opened the pull request:
## Changes

Adds Hono as a framework. Primarily focused on using Hono on the server side. 

## Checklist

- [x] Words are spelled using American English
- [x] Titles are in [sentence case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case)
- [x] Feature names are in **[sentence case too]([https://apastyle.apa.org/style-grammar-guidelines/capitalization/title-case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case))**
- [ ] Use relative URLs for internal links
- [ ] If I moved a page, I added a redirect in `vercel.json`
- [ ] Remove this template if you're not going to fill it out!

## Article checklist

- [ ] I've added (at least) 3-5 internal links to this new article
- [ ] I've added keywords for this page to the rank tracker in Ahrefs
- [ ] I've checked the preview build of the article
- [ ] The date on the article is today's date
- [ ] I've added this to the relevant "Tutorials and guides" docs page (if applicable)

ivanagas commented on contents/docs/libraries/hono.mdx:
````diff
@@ -0,0 +1,77 @@
+
+Then, create a [middleware](https://hono.dev/docs/guides/middleware) to capture events for your routes.
+
+```js file="middleware.js"
````
Needs imports?

gewenyu99 replied:
Added

ivanagas commented on contents/docs/libraries/hono.mdx:
````diff
@@ -0,0 +1,77 @@
+Next, apply this middleware to your app. This middleware will be called on every request handled by Hono,
+denoted by the wildcard `*`.
+
+```ts file=index.ts
````
Needs imports?

gewenyu99 replied:
Added

ivanagas commented on contents/docs/libraries/hono.mdx:
```diff
@@ -0,0 +1,77 @@
+In your routes, you can pass information to be captured by your middleware using [context `set()` and `get()` pair](https://hono.dev/docs/api/context#set-get)
+
+## Error tracking
+Hono uses [`app.onError`](https://hono.dev/docs/api/exception#handling-httpexception) for root-level
+error handling. You can take advantage of this for [Error tracking](/docs/error-tracking).
```
```suggestion
Hono uses [`app.onError`](https://hono.dev/docs/api/exception#handling-httpexception) for root-level error handling. You can take advantage of this for [Error tracking](/docs/error-tracking).
```

ivanagas commented on contents/docs/libraries/hono.mdx:
Which Hono template are you using for this? I think it is worth specifying because with Node, it doesn't seem to work as it is now.

gewenyu99 replied:
Huh, I'm using Node as well. Interesting

gewenyu99 commented:
@ivanagas Take another look, turns out the code is the same on cloudflare, vercel, bun, etc.
- Adds a note about suggesting they follow first with a fresh node.js hono template
- Added special note about cloudflare pages, which is the only place with a minor difference with middleware
- Added imports.
- Moved everything into index.ts for easier following.

ivanagas commented on contents/docs/libraries/hono.mdx:
```diff
@@ -0,0 +1,96 @@
+
+PostHog makes it easy to get data about usage of your [Hono](https://hono.dev/) app. Integrating PostHog into your app enables analytics, custom events capture, feature flags, and more.
+
+This guide walks you through integrating PostHog with your Hono app running on the [Node.js](https://hono.dev/docs/getting-started/nodejs) runtimes. You can hook into the same middleware and `onError` for other runtimes to capture events.
```
```suggestion
This guide walks you through integrating PostHog with your Hono app running on the [Node.js](https://hono.dev/docs/getting-started/nodejs) runtime, but has also been tested with Cloudflare Workers, Vercel, and Bun. You can hook into the same middleware and `onError` for other runtimes to capture events.
```

ivanagas commented on contents/docs/libraries/hono.mdx:
````diff
@@ -0,0 +1,96 @@
+npm install --save posthog-node
+```
+
+Then, create a [middleware](https://hono.dev/docs/guides/middleware) to capture events for your routes. Remember to export your [project API key](https://us.posthog.com/settings/project#variables) as an environment variable.
````
```suggestion
Then, create a [middleware](https://hono.dev/docs/guides/middleware) to capture events for your routes. Remember to **export** your [project API key](https://us.posthog.com/settings/project#variables) as an environment variable.
```

ivanagas commented on contents/docs/libraries/hono.mdx:
```diff
@@ -0,0 +1,96 @@
+
+const posthogServerMiddleware = createMiddleware(async (c, next) => {
+  const { POSTHOG_PUBLIC_KEY } = env<{ POSTHOG_PUBLIC_KEY:string }>(c)
+  const posthog = new PostHog(POSTHOG_PUBLIC_KEY, { host: 'https://us.i.posthog.com' })
```
ph_client_api_host will autofill with the correct value for eu vs us
```suggestion
  const posthog = new PostHog(POSTHOG_PUBLIC_KEY, { host: '<ph_client_api_host>' })
```

ivanagas commented on contents/docs/libraries/hono.mdx:
```diff
@@ -0,0 +1,96 @@
+
+<CalloutBox icon="IconInfo" title="Shutdown">
+
+Make sure to always call `posthog.shutdown()` after capturing events from the server-side. PostHog queues events into larger batches, and this call forces all batched events to be flushed immediately.
```
```suggestion
Make sure to always call `posthog.shutdown()` after capturing events from the server-side. PostHog queues events into larger batches. This call forces all batched events to be flushed immediately.
```

ivanagas commented on contents/docs/libraries/hono.mdx:
```diff
@@ -0,0 +1,96 @@
+
+app.onError((err, c) => {
+  const { POSTHOG_PUBLIC_KEY } = env<{ POSTHOG_PUBLIC_KEY:string }>(c)
+  const posthog = new PostHog(POSTHOG_PUBLIC_KEY, { host: 'https://us.i.posthog.com' })
```
```suggestion
  const posthog = new PostHog(POSTHOG_PUBLIC_KEY, { host: '<ph_client_api_host>' })
```

ivanagas commented on contents/docs/libraries/hono.mdx:
```diff
@@ -0,0 +1,96 @@
+
+## Next steps
+
+For any technical questions for how to integrate specific PostHog features into Hono, have a look at our [Node](/docs/libraries/node) SDK docs for concepts such as:
```
```suggestion
To read more about how to integrate specific PostHog features into Hono, have a look at our [Node SDK docs](/docs/libraries/node) for concepts such as:
```

ivanagas commented on contents/docs/libraries/hono.mdx:
```diff
@@ -0,0 +1,96 @@
+## Next steps
+
+For any technical questions for how to integrate specific PostHog features into Hono, have a look at our [Node](/docs/libraries/node) SDK docs for concepts such as:
+- [Capture custom events, identify users, and more.](/docs/libraries/node#capturing-events)
+- [Feature flags including variants and payloads.](/docs/libraries/node#feature-flags)
```
Identifying isn't that important in this context, identify broadly is more of a frontend concept
```suggestion
- [Capturing custom events, setting properties, and more.](/docs/libraries/node#capturing-events)
- [Setting up feature flags including variants and payloads.](/docs/libraries/node#feature-flags)
```

ivanagas reviewed (approved):
some tweaks, but looks good :)
